fbw-a380x/src/systems/instruments/src/PFD/FMA.tsx

```diff
@@ -1429,6 +1503,43 @@ class BC3Cell extends DisplayComponent<{ isAttExcessive: Subscribable<boolean>;
         this.tdReached = tdr;
         this.fillBC3Cell();
       });
+
+    sub
+      .on('altitudeAr')
+      .whenChanged()
+      .handle((alt) => {
+        this.baroCorrectedAltitude = alt.value;
+        this.fillBC3Cell();
```

Comment:
This will be quite expensive as the altitude is constantly changing and fillBC3Cell allocates a lot of strings. It would be better to compute the actual warning state in a subscribable and then subscribe fillBC3Cell to that.

Reply:
I've not worked with MappedSubject yet, so I have a few questions, sorry 😅 

Is the correct approach to add these under the other defined properties of `BC3Cell`

```
private isActiveModeVsFpa = false;

private readonly sub = this.props.bus.getSubscriber<PFDSimvars & Arinc429Values & SimplaneValues>();

private readonly activeVerticalMode = ConsumerSubject.create(this.sub.on('activeVerticalMode'), 0);

private readonly activeVerticalModeMapped = MappedSubject.create(([activeVerticalMode]) => {
  if (activeVerticalMode === VerticalMode.FPA || activeVerticalMode === VerticalMode.VS) {
    return true;
  }
  return false;
}, this.activeVerticalMode) as Subscribable<boolean>;
```
then define this function in `onAfterRender`

```
this.activeVerticalModeMapped.sub((val) => {
      this.isActiveModeVsFpa = val;
    });
```

and pass `this.isActiveModeVsFpa` as a parameter to `getBC3Message` in `fillBC3Cell` ?

I will of course pass the other variables into the MappedSubject after, this is just for the question.

Also, the


```
sub
      .on('altitudeAr')
      .whenChanged()
      .handle((alt) => {
        this.altitude = alt.value;
        this.fillBC3Cell();
      });
```

I put into the `onAfterRender` function initially is something with Arinc429, so I had to use .value to get the altitude I needed, but I am not able to do the same with the ConsumerSubject. If it's correct, how do I get the value if feet ?

Thanks!

Reply:
Nevermind, I've just pushed a commit where I tried to do this. I've not worked with MappedSubject yet so sorry if it's a total nonsense, just check it out when you have time please :)